### PR TITLE
[EDS-563] Export department and title information in JSON data.

### PR DESCRIPTION
**Pull request by tomthorogood**
This is a sub-PR of #51, for the action items: 

>  - [x] Return department in output models (investigating the other issue would be easier if the JSON had that... 😂)
>    - [x] Add a test to determine its inclusion in output models.

It was a medium amount of work that really ought to have been its own jira, but live and learn.

When merged, this will be merged into the EDS-563 branch (which will update that PR), after which I'll work on the remaining action items.

See individual commits for details, I tried to break them out into sensible chunks.

The tl;dr is that department data is now exported and visible via the JSON get:

```
/search?name="cauce"
```

```
// . . . 
      "description": "Last name is \"cauce\"", 
      "numResults": 1, 
      "populations": {
        "employees": {
          "numResults": 1, 
          "people": [
            {
              "departments": [
                {
                  "department": "University of Washington", 
                  "title": "President"
                }, 
                {
                  "department": "Psychology", 
                  "title": "Professor"
                }, 
                {
                  "department": "American Ethnic Studies", 
                  "title": "Professor"
                }, 
                {
                  "department": "College of Education", 
                  "title": "Adjunct Professor"
                }, 
                {
                  "department": "Gender, Women and Sexuality Studies", 
                  "title": "Adjunct Professor"
                }, 
                {
                  "department": "University of Washington", 
                  "title": "President"
                }
              ], 
              "email": "president@uw.edu", 
              "href": "L2lkZW50aXR5L3YyL3BlcnNvbi84OUQ5NkRGODZBN0MxMUQ1QTRBRTAwMDRBQzQ5NEZGRS9mdWxsLmpzb24=", 
              "name": "Ana Mari Cauce", 
              "phoneContacts": {
                "faxes": [], 
                "mobiles": [], 
                "pagers": [], 
                "phones": [
                  "+1 206 543-5010"
                ], 
                "touchDials": [], 
                "voiceMails": []
              }
            }
          ], 
// . . . 
```

### CODE REVIEW

**Comment by tomthorogood** [comment]
ironically, the "Applied Mathematics" department search from #51 breaks this, presumably because of something funny in the data...

**Comment by goulter** [comment]
Oh, now that is priceless. Onwards 👍

**Comment by tomthorogood** [comment]
Looks like the existing directory doesn't include entries like that, so i can ignore them too. i'll post that patch shortly.

**Comment by goulter** [comment]
Ya, I just did a search on "None" for dept, also tried an empty string -> both, no results.
Interesting, the None might be substituted value, but why wouldn't it be a string? I know thats not our concern, anyway I just find that curious. Curiousity ???'d the programmer.

**Comment by tomthorogood** [comment]
I thought I added a comment before but it's gone now. Anyway: If you searched for None, you were searching for the string "None," but the value is not a string, it is a null value. There is no way (nor should there be) to search for `None` from the user interface.

**Comment by goulter** [comment]
Do you have the name of the person that generates
`EmployeePosition(title='Associate Professor', department=None)]`?

**Comment by tomthorogood** [comment]
I don't want to post specific names and netids associated with data validation in a public github thread.

**Review by goulter — APPROVED** [review]
I think I understand. This is just code I'm still getting used to.

File: husky_directory/services/translator.py [hunk 1]
```diff
@@ -170,6 +170,7 @@ def filter_(person_) -> bool:
                         title=student.directory_listing.class_level, department=dept
                     )
                     for dept in student.directory_listing.departments
+                    if dept  # Ignore data with holes in it.
```
[the remark this answers is not in the document]
Comment (tomthorogood) [reply husky_directory/services/translator.py:173]: yes, it's in brackets/parentheses. this is a listcomp.

[the remark this answers is not in the document]
Comment (tomthorogood) [reply husky_directory/services/translator.py:173]: (it would be stylistically incorrect to indent here.)

**Pull request merged**